src/main/io/gps.h

```diff
@@ -145,9 +329,19 @@ typedef struct gpsData_s {
 
     uint8_t ackWaitingMsgId;        // Message id when waiting for ACK
     uint8_t ackTimeoutCounter;      // Ack timeout counter
-    ubloxAckState_e ackState;
-    bool ubloxUsePVT;
-    bool ubloxUseSAT;
+    ubloxAckState_e ackState;       // Ack State
+    uint8_t updateRate;
+    bool ubloxM7orAbove;
+    bool ubloxM8orAbove;
+    bool ubloxM9orAbove;
+    bool ubloxUsingFlightModel;    // false = Acquire model, true = Flight model
+    bool satMessagesDisabled;
+#ifdef USE_GPS_UBLOX
+    uint32_t gpsNavSolIntervalMs;    // interval between successive UBX nav solutions in ms
+    uint32_t lastNavSolTs;      // time stamp of last UBCX message.  Used to calculate message delta
+    ubloxVersion_e unitVersion;     // detected UNIT version
```

Comment:
```suggestion
    ubloxVersion_e ubloxVersion;     // detected ublox version
```

Reply:
Changed to platformVersion

Reply:
But "platform" is too universal. This is specifically the Ublox version. Remember that we also support NMEA.

Reply:
u-blox calls their product versions (M7, M8, M9) "platform" versions. This is the combination of hardware and firmware they package together under those product platform names. They are mostly, but not not universally, consistent with this terminology in all their docs.

We are translating to those platform versions and we use the M7, M8, M10 terminology in the code, so I think we should use the correct name for that.

We derive the platform version from the hardware version reported by the module. This is the version number of the chipset. The hardware version is a string like "00040005". There is a 1:1 mapping between hardware version and platform version, in other words when they change their chipset they change the platform.

There is also a software/firmware version. Just like what we'd think, it's the code release version. The reported software version includes where the firmware is running from, for example "ROM SPG 5.10 (c00d69)" is firmware running from ROM. The firmware version on its own is just the version part, "SPG 5.10" meaning Standard Precision GNSS version 5.10. There can be multiple firmware versions released for a platform/hardware version. Not a whole lot of use to us, other than for info about the module.

Then there is the protocol version. Each protocol, NMEA and UBX, has its own versioning.

For NMEA, the versions are the open standard versions like 0183. I don't think they've been updated for some time, and afaik, the module doesn't report the NMEA version. u-blox wants us to consider their UBX protocol version as also their NMEA protocol version, but that's not really correct in terms of the NMEA standard.

The UBX protocol version is reported in the extension field as PROTVER, and is a number like 34.10. The UBX protocol version is a bit more interesting to us, since this is really what determines when certain UBX messages become supported or deprecated or removed. But as it turns out, these things only happen for protocol versions that change at the same time the hardware/platform version changes, so the hardware/platform version works as a proxy for UBX protocol version. If we intend to support more than just the Standard Precision GNSS modules, we might need to reconsider this.

Probably way more than you wanted to know, but there is a specific reason I precisely chose the term "platformVersion" to match what it represents in u-blox terminology. In the u-blox context, it's not a universal term, and I wanted someone trying to reference from the code to the u-blox docs to be able to match back.